src/main/java/com/snowplowanalytics/redash/model/User.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2018 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.redash.model;
+
+import java.util.List;
+import java.util.Objects;
+
+public class User extends BaseEntity {
+    private List<Integer> groups;
```

Comment:
Groups should be a separate entity - the same Group can have User members and can have Data Sources. A User has no awareness of whether it belongs to a Group. Likewise with a Data Source. This is also clear in the API calls where you have `baseUrl + GROUPS_URL_PREFIX` - the operations are performed at the Group level, not the individual attributes that a Group is made up of.

Reply:
It looks like this has been partially implemented in the UserGroup class - this should be generalised so that a Group has awareness of both Users and Data Sources.